api/v0/finder/models/models.go

```diff
@@ -87,37 +88,49 @@ func UnmarshalResp(b []byte) (*Response, error) {
 	return r, err
 }
 
-// PopulateResp reads from indexer engine to populate a response from a list of CIDs.
-func PopulateResp(e *indexer.Engine, cids []cid.Cid) (*Response, error) {
-	out := Response{
-		Cids:      make([]CidData, len(cids)),
-		Providers: make([]ProviderData, 0),
-	}
-	num := 0
+// PopulateResponse reads from indexer engine to populate a response from a
+// list of CIDs.
+//
+// TODO: This should be relocated to a place common to all finder handlers, but
+// not under /api/
+func PopulateResponse(engine *indexer.Engine, registry *providers.Registry, cids []cid.Cid) (*Response, error) {
+	cidResults := make([]CidData, 0, len(cids))
+	var providerResults []ProviderData
+	providerSeen := map[peer.ID]struct{}{}
 
 	for i := range cids {
-		v, f, err := e.Get(cids[i])
+		values, found, err := engine.Get(cids[i])
 		if err != nil {
-			// If error and the request is not for a batch, continue without
-			// sending an error as if the errored CID wasn't found
-			if len(cids) < 1 {
-				return nil, err
-			}
+			return nil, fmt.Errorf("failed to query cid %q: %s", cids[i], err)
```

Comment:
Just a note: do we expect the `Get` for a small set of CIDs of a batch to be able to fail when querying the indexer engine? If this is the case I would not return an error right away, and continue as if the CID wasn't found. I think a batch query shouldn't err just because a one CID in the batch has err. If we don't expect this to happen, I am happy with how it is now.

Reply:
Actually, I think this is fine how it is. If a `Get` fails it is probably because something is wrong with the indexer-core and we should be able to trace it. Nevermind :)